dev-client/components/SiteMap.tsx

```diff
@@ -0,0 +1,47 @@
+import Mapbox, {Camera, MarkerView, UserLocation} from '@rnmapbox/maps';
+import React, {useEffect, useRef} from 'react';
+import {type DisplaySite} from '../datatypes/sites';
+import {type Position} from '@rnmapbox/maps/lib/typescript/types/Position';
+import {PermissionsAndroid} from 'react-native';
+import {IconButton, Icon} from 'native-base';
+// TODO: This vector icons module appears to be missing types
+import VIcon from 'react-native-vector-icons/MaterialIcons';
+
+type SiteMapProps = {
+  sites: DisplaySite[];
+  center: Position;
+};
+
+// REVIEW: This is a bad name, any suggestions?
+export default function SiteMap({sites, center}: SiteMapProps): JSX.Element {
+  const camera = useRef<Camera>(null);
+
+  useEffect(() => {
+    camera.current?.setCamera({
+      centerCoordinate: center,
+    });
+  }, [center]);
+
+  return (
+    <Mapbox.MapView
+      style={{
+        flex: 1,
+      }}>
+      <Camera ref={camera} />
+      <UserLocation />
+      {sites.map(site => {
+        return (
+          <MarkerView coordinate={[site.lon, site.lat]} key={site.key}>
```

Comment:
question: did you look into alternative implementations than `MarkerView`? (see https://github.com/rnmapbox/maps/blob/main/docs/Annotations.md). i think `MarkerView` is fine for now and we can just swap it out if we need to, just curious if you tried any of the others as i think we may need to upgrade from `MarkerView` eventually. (e.g. if we want to do point clustering without writing the clustering algorithm ourselves i don't think we could use `MarkerView`)

Reply:
No, good point. I'll add this as another issue.